lib/config/error-codes/throw.js

```diff
@@ -72,25 +84,84 @@ function throwError (domain, subdomain, error, ...placeholders) {
+ * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
+ * @param  {string} error - Error name: (eg: 'index_not_found')
+ * @param  {...any} placeholders - Placeholders value to inject in error message
+ */
+function getErrorFrom (errorSource, domain, subdomain, error, ...placeholders) {
 
- * The function have now the following signature:
+  const
+    errorName = `${domain}.${subdomain}.${error}`,
+    kuzzleError = _.get(errorCodes, `${domain}.subdomains.${subdomain}.errors.${error}`);
+
+  if (! kuzzleError) {
+    return getErrorFrom(
+      errorSource,
+      'internal',
+      'unexpected',
+      'unknown_error',
+      ...placeholders);
+  }
+
+  const code =
+    (errorCodes[domain].code >> 24) |
+    (errorCodes[domain].subdomains[subdomain].code >> 16) |
+    errorCodes[domain].subdomains[subdomain].errors[error].code;
+
+  errorSource.message = format(kuzzleError.message, ...placeholders, errorSource.message);
+
+  if (kuzzleError.class === 'PartialError') {
+    const body = placeholders.slice(-1)[0];
+    return new errors[kuzzleError.class](errorSource, body, errorName, code);
+  }
+
+  return new errors[kuzzleError.class](errorSource, errorName, code);
+}
+
+/**
+ * Construct and throw the corresponding error
+ * @param  {string} errorSource - non KuzzleError from which the KuzzleError will derivate
+ * @param  {string} domain - Domain (eg: 'external')
+ * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
+ * @param  {string} error - Error name: (eg: 'index_not_found')
+ * @param  {...any} placeholders - Placeholders value to inject in error message
+ */
+function throwFrom (errorSource, domain, subdomain, error, ...placeholders) {
+  throw getErrorFrom(errorSource, domain, subdomain, error, ...placeholders);
+}
+
+/**
+ * Wrap getError, getErrorFrom, throw and throwFrom
+ * with the provided domain and subdomain.
+
+ * The functions have now the following signature:
  *   - getError(error, ...placeholders)
  *   - throw(error, ...placeholders)
- *
+ *   - getErrorFrom: (errorSource, error, ...placeholders)
+ *   - throwFrom: (errorSource, error, ...placeholders)
+ 
+ * @param  {string} errorSource - non KuzzleError from which the KuzzleError will derivate
  * @param  {string} domain - Domain (eg: 'external')
  * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
  */
 function wrap (domain, subdomain) {
   return {
     getError: (error, ...placeholders) => getError(domain, subdomain, error, ...placeholders),
-    throw: (error, ...placeholders) => throwError(domain, subdomain, error, ...placeholders)
+    throw: (error, ...placeholders) => throwError(domain, subdomain, error, ...placeholders),
+    getErrorFrom: (errorSource, error, ...placeholders) => getErrorFrom(errorSource, domain, subdomain, error, ...placeholders),
+    throwFrom: (errorSource, error, ...placeholders) => throwError(errorSource, domain, subdomain, error, ...placeholders)
   };
 }
 
 module.exports = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Group all shorthand properties at either the beginning or end of this object declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3499)